airbyte-integrations/connectors/destination-gcs/src/main/java/io/airbyte/integrations/destination/gcs/GCSDestinationConfig.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.integrations.destination.gcs;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+public class GCSDestinationConfig {
+
+  private final String bucketName;
+  private final String bucketPath;
+  private final String bucketRegion;
+  private final String accessKeyId;
+  private final String secretAccessKey;
+  private final String projectId;
+  private final GCSFormatConfig formatConfig;
+
+  public GCSDestinationConfig(
+                             String bucketName,
+                             String bucketPath,
+                             String bucketRegion,
+                             String accessKeyId,
+                             String secretAccessKey,
+                             String projectId,
+                             GCSFormatConfig formatConfig) {
+    this.bucketName = bucketName;
+    this.bucketPath = bucketPath;
+    this.bucketRegion = bucketRegion;
+    this.accessKeyId = accessKeyId;
+    this.secretAccessKey = secretAccessKey;
+    this.projectId = projectId;
+    this.formatConfig = formatConfig;
+  }
+
+  public static GCSDestinationConfig getGCSDestinationConfig(JsonNode config) {
+    return new GCSDestinationConfig(
+        config.get("gcs_bucket_name").asText(),
+        config.get("gcs_bucket_path").asText(),
+        config.get("gcs_bucket_region").asText(),
+        config.get("access_key_id").asText(),
+        config.get("secret_access_key").asText(),
+        config.get("project_id").asText(),
```

Comment:
`project_id` is an optional field according to `spec.json`. So this line may throw NPE if `project_id` is not specified.

Reply:
I think I ended up not using `project_id` at all so I will just remove it completely from here and `spec.json`. Need to double check though